src/api/assistant.rs

```diff
@@ -0,0 +1,105 @@
+use crate::ratectl::{
+    SlackApiMethodRateControlConfig, SlackApiRateControlLimit, SlackApiRateControlSpecialLimit,
+    SLACK_TIER4_METHOD_CONFIG,
+};
+use crate::{
+    ClientResult, SlackAssistantPrompt, SlackChannelId, SlackClientHttpConnector,
+    SlackClientSession, SlackTs,
+};
+use lazy_static::lazy_static;
+use rsb_derive::Builder;
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+impl<'a, SCHC> SlackClientSession<'a, SCHC>
+where
+    SCHC: SlackClientHttpConnector + Send,
+{
+    /// https://api.slack.com/methods/assistant.threads.setStatus
+    pub async fn assistant_threads_set_status(
+        &self,
+        req: &SlackApiAssistantThreadsSetStatusRequest,
+    ) -> ClientResult<SlackApiAssistantThreadsSetStatusResponse> {
+        self.http_session_api
+            .http_post(
+                "assistant.threads.setStatus",
+                req,
+                Some(&ASSISTANT_THREAD_SET_STATUS_SPECIAL_LIMIT_RATE_CTL),
+            )
+            .await
+    }
+
+    /// https://api.slack.com/methods/assistant.threads.setSuggestedPrompts
+    pub async fn assistant_threads_set_suggested_prompts(
+        &self,
+        req: &SlackApiAssistantThreadsSetSuggestedPromptsRequest,
+    ) -> ClientResult<SlackApiAssistantThreadsSetStatusResponse> {
```

Comment:
`SlackApiAssistantThreadsSetStatusResponse` this seems a misleading type name. Should we create another empty structure to be consistent?

Reply:
Fair enough copy paste fail xD